apps/httpd/main.cc

```diff
@@ -80,11 +81,19 @@ int main(int ac, char** av) {
     app.add_options()("prometheus_address", bpo::value<sstring>()->default_value("0.0.0.0"), "Prometheus address");
     app.add_options()("prometheus_prefix", bpo::value<sstring>()->default_value("seastar_httpd"), "Prometheus metrics prefix");
 
-    return app.run_deprecated(ac, av, [&] {
+    return app.run(ac, av, [&] {
         return seastar::async([&] {
             seastar_apps_lib::stop_signal stop_signal;
             auto&& config = app.configuration();
             httpd::http_server_control prometheus_server;
+            bool prometheus_started = false;
+
+            auto stop_prometheus = defer([&] () noexcept {
+                if (prometheus_started) {
```

Comment:
`sharded<Service>::stop()` is a no-op. so probably we can dispense with this `prometheus_started` flag and use `deferred_stop` instead?

Reply:
It's not clean to stop() something that isn't started and rely on implementation details.

Reply:
can we document this behavior in `shared<>` that `sharded<Service>::stop()` is a no-op if the sharded service is not started at all? so it won't be implementation details anymore, but a feature. just like `delete null` is a no-op.

Reply:
IMO it encourages bad coding practices.